components/controls/notifications/NotificationSettings.tsx

```diff
@@ -0,0 +1,92 @@
+import { useQuery } from '@tanstack/react-query'
+
+import NotificationControl from '@/components/controls/notifications/NotificationControl'
+import Field from '@/components/shared/Field'
+import { useTranslation } from '@/hooks/useTranslation'
+import { clientApi } from '@/modules/backend/client-api'
+import { SaveUserSettingsDto } from '@/modules/backend/openapi-generated'
+
+const NotificationSettings = () => {
+  const t = useTranslation('Settings')
+
+  const { data: response, refetch } = useQuery({
+    queryKey: ['NotificationSetting'],
+    queryFn: () => clientApi.usersControllerGetUserSettings(),
+  })
+
+  if (!response?.data) {
+    return null
+  }
+
+  // TODO catch error
+  const saveNotifications = async (value: SaveUserSettingsDto) => {
+    try {
+      await clientApi.usersControllerSaveUserSettings(value)
+      await refetch()
+    } catch (error) {
+      console.log(error)
+    }
+  }
```

Comment:
When I switch the switch e.g. from off to on, it firstly turns to on for a brief moment then off again and then finally on, this is probably caused by the fact that the switches are directly bound to the server data
I would use the "Optimistic updates" pattern here and add an intermediate front end state layer, the switch would directly switch to the desired value, and we would only handle an erroneous response (a mismatch between the backend and frontend data)

Reply:
Definitely, creating new issue for this.